include/matx/core/tensor_utils.h

```diff
@@ -56,23 +62,44 @@ namespace matx
     return 0;
   }
 
+
+  /**
+   * @brief finds the size of the largest dim of the tensor
+   *core/tensor_utils.h
+   * @param op Operator
+   * @return size of largest dim
+   */
+  template <typename Op>
+  index_t LargestDimSize(const Op &op) {
+    index_t maxSize = 0;
```

Comment:
Slightly faster to set this to Size(0)  and iterate from 1..Rank()

Reply:
optimized to iterate from 1st element